src/lib.rs

```diff
@@ -254,6 +254,9 @@ impl DomainExtractor {
             );
         }
 
+        if url_str.len() > 255 {
+            return Err(PyValueError::new_err("Invalid domain detected"));
```

Comment:
I suggest being more specific here as per the other example - `url is invalid: longer than 255 characters` / `url is invalid: too long` / etc